Add PlantUML processing and example

awwaiid opened the pull request:
Experimental in-tree generated diagrams

awwaiid commented:
This is now ready for review/merge. It ended up with:

* A new github action that watches for new diagrams, renders them, and adds them in another commit
* A demonstration of embedded diagrams is included at https://github.com/rubyforgood/human-essentials/blob/d1c20048fd8b5decc78f64cb71746eea9c71a56c/doc/architecture/overview.md
* This demonstration works out how we can do inline diagrams

dorner commented on doc/architecture/overview.md:
```diff
@@ -0,0 +1,205 @@
+
+This application is multi-tenant -- that is, each Diaper Bank (used interchangeably with **Organization**) has its own templated "section" of the application, and may act in that section without concern that its changes will affect other organizations (and vice versa).
+
+When a user is signed-in, they are automatically "isolated" to their organizational space, as indicated by the URL (which features a short-code of their organization in it).
```
With [this PR](https://github.com/rubyforgood/human-essentials/pull/3386) this won't really be true any more - the organization ID in the URL isn't relevant, it's the current role of the user. I'm not even sure that referring to the app as "multi-tenant" is true; usually a multi-tenanted app refers to the database structure (where each client has their own table or set of tables).

awwaiid replied:
I thought multi-tenant meant multiple-organizations on a single-db instead of each org having their own db. But really I copy/pasted this from a previous thing, so I'm simplifying :)

dorner reviewed (changes requested):
One comment on the technical side - I think @cielf needs to look at this from an app experience perspective. From what I know it looks really good!

cielf commented:
Sorry I haven't gotten to this yet -- it feels like something I should look at when I'm fresh!

cielf reviewed (changes requested):
Overall, I think this is going to be very helpful to folk.   We also just added announcements, and will be adding counties -- do those get a place here too?

cielf commented:
@awwaiid  We'll also want to put something into the README to direct people to this.

awwaiid commented:
Consider adding `announcements` or other concepts

awwaiid commented on doc/architecture/overview.md:
```diff
@@ -0,0 +1,205 @@
+
+### Intake
+
+A Diaperbank will acquire inventory primarily through one of two methods: a **Donation** or a **Purchase**. Purchases are straightforward; the Diaperbank spends its own money to purchase inventory. Donations can be received through a few different means.
```
I went through and switched most of the Diaper references to Essentials

awwaiid commented on doc/architecture/overview.md:
```diff
@@ -0,0 +1,205 @@
+Distributions can be exported as PDFs, which Banks can use as printable manifests for the packages sent to the Community Partner.
+
+Community Partners then dispense the items they receive to the clients / families.
+
```
Done, and added a note as to what a Partner typically is

awwaiid commented on doc/architecture/overview.md:
````diff
@@ -0,0 +1,205 @@
+```
+</details>
+
+Partners may submit requests to a Bank.
````
Included something into the general definition, but I'll leave the family-vs-child-vs-qty details for another time.

awwaiid commented:
I think I've addressed most of the feedback. If it looks tolerable for a draft then I'd like to merge it. As part of preparing for RFG-2023 I will submit further PRs for README cleanups which will link to this.

dorner reviewed (approved):
I think it's way better than what we have now! Let's get it merged and make it better as needed. @cielf any objections?

cielf commented:
No objections.
